### DISCUSSION TITLE
How to change separator character in the document title?

### DISCUSSION THREAD

**Discussion by glow12121** [datenstrom/community#950] [Ask a question] [answered]
Hi,

Is it possible to change the separator character in the website's name, the one displayed in the browser's title bar? It is normally something in the way of: "Page title (taken from `pagename.md`) - Sitename (taken from `yellow-system.ini`)". Here, the separator is ` - ` (a hyphen, plus two spaces before and after it).

**Comment by GiovanniSalmeri** [comment]
A quick and dirty solution is to modify lines 2987-2988 in `system/extensions/core.php`: it is there that the `titleHeader` of the page is created. Unfortunately this solution is **very** dirty: at each update you will lose your modification. A cleaner solution is to change line 4 of `system/layouts/header.html`. Instead of using the `titleHeader` you can build a string however you wish. Hope this helps!

**Reply by glow12121** [reply, marked as answer]
Thank you for the comments! Line 4 of `header.html` has this code: `<title><?php echo $this->yellow->page->getHtml("titleHeader") ?></title>` How can I only change the separator character here?

Should I just change it to something like this (`" | "` instead of `" - "`)?

```
<title><?php echo 
$titleHeader = ($this->location==$this->yellow->content->getHomeLocation($this->location)) ?
                $this->get("sitename") : $this->get("title")." | ".$this->get("sitename");
?></title>
```

It seems to be working alright.

**Comment by markseuffert** [comment]
Set `TitleHeader` in the [page settings](https://github.com/annaesvensson/yellow-core#settings-page) at the top of a page. Here's an example content file:

```
---
Title: For people who like to drink espresso
TitleHeader: For people who like to drink espresso | Glow1212
---
This is an example page.
```

This solution works on a per-page-basis. Good when you want to adjust the title of a few pages. Mind-numbing terrible when you have to adjust the title of all pages. Fortunately the API for developers lets you automate things. As Giovanni kindly explained, you could use a layout file or a small extension to change the HTML title element of all pages on your website.

**Reply by glow12121** [reply]
Thank you for the comments! I've replied above to your colleague Giovanni.
